pkg/etcd/etcd.go

```diff
@@ -495,11 +501,68 @@ func GetAdvertiseAddress(advertiseIP string) (string, error) {
 
 // newCluster returns options to set up etcd for a new cluster
 func (e *ETCD) newCluster(ctx context.Context, reset bool) error {
-	return e.cluster(ctx, reset, executor.InitialOptions{
+	err := e.cluster(ctx, reset, executor.InitialOptions{
 		AdvertisePeerURL: fmt.Sprintf("https://%s:2380", e.address),
 		Cluster:          fmt.Sprintf("%s=https://%s:2380", e.name, e.address),
 		State:            "new",
 	})
+	if err != nil {
+		return err
+	}
+	if err := e.migrateFromSQLite(ctx); err != nil {
+		return fmt.Errorf("failed to migration content from sqlite to etcd: %w", err)
+	}
+	return nil
+}
+
+func (e *ETCD) migrateFromSQLite(ctx context.Context) error {
+	_, err := os.Stat(sqliteFile(e.config))
+	if os.IsNotExist(err) {
+		return nil
+	} else if err != nil {
+		return err
+	}
+
+	logrus.Infof("Migrating content from sqlite to etcd")
+
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+
+	_, err = endpoint2.Listen(ctx, endpoint2.Config{
```

Comment:
Are we leaking this listener on the kine socket when we start a new cluster with sqlite data available?

Reply:
That's the reason for the `defer cancel()` above.  The only way to close the listener is to close the context.

Reply:
Ah, pardon the low-caffeine "review"